### ISSUE TITLE
Changes to upstream icestorm break icebram replacement

### ISSUE THREAD

**Issue by cibomahto** [tillitis/tillitis-key1#91]
After setting up a new build environment by either following the toolchain setup or building a fresh Docker image, the application gateware build fails to replace the bram correctly:

```
$make prog_flash
...
icebram -v bram_fw.hex firmware.hex < application_fpga.asc > application_fpga.asc.tmp
Loaded pattern for 32 bits wide and 1536 words deep memory.
Extracted 192 bit slices from from/to hexfile data.
Found 21 initialized bram cells in asc file.
No memory instances were replaced.
make: *** [Makefile:201: application_fpga.bin] Error 1
```

The issue appears to be somewhere between commit 45f5e5f3889afb07907bab439cf071478ee5a2a5 and d20a5e9001f46262bf0cef220f1a6943946e421d in incestorm: https://github.com/YosysHQ/icestorm/commits/master

Reverting icestorm back to commit 45f5e5f3889afb07907bab439cf071478ee5a2a5 seems to fix the issue, so let's pin icestorm to that commit.

It's possible that upstream yosys/nextpnr have evolved to be compatible with the changes to icebram, so this could be reconsidered when the other tools are also updated.

**Comment by cibomahto** [comment]
dockerfile addressed in: https://github.com/tillitis/tillitis-key1/commit/938f2b16d7b685012ab130fede7f03bb8cc1747b

**Comment by cibomahto** [comment]
toolchain setup addressed in: e04b502
